mem/buffer_pool.go

```diff
@@ -32,6 +32,9 @@ type BufferPool interface {
 	Get(length int) *[]byte
 
 	// Put returns a buffer to the pool.
+	//
+	// The returned pointer must hold a prefix of the buffer obtained via [Get]
```

Comment:
The use of "returned" here is confusing since the function doesn't return anything.

```suggestion
	// The provided pointer must hold a prefix of the buffer obtained via [Get]
```

Or 

```suggestion
	// The pointer to be returned must hold a prefix of the buffer obtained via [Get]
```

Or name the parameter, e.g. `buf`, and say `buf must hold a prefix of ...`.

Also, maybe "hold" -> "be"?